frontend/www/js/omegaup/arena/navigation.test.ts

```diff
@@ -92,17 +117,19 @@ const navbarProblems: types.NavbarProblemsetProblem[] = [
 describe('navigation.ts', () => {
   describe('navigateToProblem', () => {
     it('Should change hash when contest alias is declared in practice mode', async () => {
-      const params: NavigationRequest = {
+      navigateToProblem({
         type: NavigationType.ForContest,
         target: vueInstance,
         runs: [],
         problems: navbarProblems,
         problem: navbarProblems[0],
-        contestAlias: 'problem_alias',
-      };
-      if (vueInstance.problemInfo) {
-        navigateToProblem(params);
-      }
+        contestAlias: 'contest_alias',
+      } as NavigationRequest);
+
+      const getLocationHash = jest
+        .fn()
+        .mockReturnValue('#problems/problem_alias');
+      expect(getLocationHash()).toEqual(`#problems/${navbarProblems[0].alias}`);
```

Comment:
esto no está verificando que `navigateToProblem` está haciendo su trabajo :( lo que se está haciendo aquí es que la primera vez que `getLocationHash()` se invoca, va a regresar `'#problems/problem_alias'`, e inmediatamente se está llamando y se está checando que haya regresado `'#problems/problem_alias'`.

en teoría se puede quitar L120-L127 y la prueba seguiría pasando :S lo que hay que hacer es inspeccionar lo que se modificó:

```suggestion
      expect(window.location.hash).toEqual(`#problems/${navbarProblems[0].alias}`);
```

si esto no funciona, habría que hacer el truco de crear un setter para el hash en `navigation.ts` para que en vez de modificar `window.location.hash = ...`, se llame `setHash(...)`, y entonces sí se puede hacer un  mock de `setHash(...)` y aseverar que se mandó llamar una vez con el parámetro esperado.

Reply:
Hice algunos cambios. Aún no estoy seguro si ya está bien.